tests/test_report.py

```diff
@@ -0,0 +1,81 @@
+import json
+import numbers
+import os
+import sys
+
+import mock
+import pytest
+
+from iopipe.config import set_config
+from iopipe.report import Report
+
+from .mock_context import MockContext
+
+
+def assert_valid_schema(obj, schema=None, path=None):
+    """Asserts that an object matches the schema"""
+
+    if not schema:
+        schema_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'schema.json')
+        with open(schema_path) as f:
+            schema = json.loads(f.read())
+
+    if not path:
+        path = []
+
+    for key in obj:
```

Comment:
So this will walk the report and raise issues if there's a type error; however the inverse (walking the schema and looking for fields in report) is what we should be doing

Reply:
Thinking we should be doing both, but as mentioned in an earlier comment, there's not currently a way in the schema.json to do much more than type checking.